hand-eye-calibration/full_hand_eye_sample.py

```diff
@@ -0,0 +1,510 @@
+
+    translation = robot_pose[:3] * 1000
+    rotvec = R.from_rotvec(robot_pose[3:])
+    transform = np.eye(4)
+    transform[:3, :3] = rotvec.as_matrix()
+    transform[:3, 3] = translation.T
+
+    return frame, transform
+
+
+def _set_camera_settings(cam: zivid.Camera):
+    """Set camera settings
+
+    Args:
+        cam: Zivid camera
+    """
+    with cam.update_settings() as updater:
+        updater.settings.iris = 20
+        updater.settings.exposure_time = datetime.timedelta(microseconds=10000)
+        updater.settings.filters.reflection.enabled = True
+        updater.settings.brightness = 1.0
+        updater.settings.filters.gaussian.enabled = 1
+        updater.settings.gain = 1
+
+
+def _read_robot_state(con: rtde):
+    """Recieve robot output recipe
+
+    Args:
+        con: Connection between computer and robot
+
+    Returns:
+        Robot state
+    """
+
+    robot_state = con.receive()
+
+    assert robot_state is not None, "Not able to recieve robot_state"
+
+    return robot_state
+
+
+def pose_from_datastring(datastring: str):
+    """Extract pose from yaml file saved by openCV
+
+    Args:
+        datastring: String of text from .yaml file
+
+    Returns:
+        Robotic pose as zivid Pose class
+    """
+
+    string = datastring.split("data:")[-1].strip().strip("[").strip("]")
+    pose_matrix = np.fromstring(string, dtype=np.float, count=16, sep=",").reshape(
+        (4, 4)
+    )
+    return zivid.handeye.Pose(pose_matrix)
+
+
+def _save_hand_eye_results(save_dir: Path, transform: np.array, residuals: list):
+    """Save transformation and residuals to folder
+
+    Args:
+        save_dir: Path to where data will be saved
+        transform: 4x4 transformation matrix
+        residuals: List of residuals
+    """
+
+    file_storage_transform = cv2.FileStorage(
+        str(save_dir / f"transformation.yaml"), cv2.FILE_STORAGE_WRITE
+    )
+    file_storage_transform.write("PoseState", transform)
+    file_storage_transform.release()
+
+    file_storage_residuals = cv2.FileStorage(
+        str(save_dir / f"residuals.yaml"), cv2.FILE_STORAGE_WRITE
+    )
+    residual_list = []
+    for res in residuals:
+        tmp = list([res.translation, res.translation])
+        residual_list.append(tmp)
+
+    file_storage_residuals.write(
+        "Per pose residuals for rotation in deg and translation in mm",
+        np.array(residual_list),
+    )
+    file_storage_residuals.release()
+
+
+def _image_count(robot_state):
+    """Read robot output register 24
+
+    Args:
+        robot_state: Robot state
+
+    Returns:
+        Number of captured images
+    """
+    return robot_state.output_int_register_24
+
+
+def _start_capture(robot_state):
+    """Read robot output register 64
+
+    Args:
+        robot_state: Robot state
+
+    Returns:
+        Boolean value that states if camera is ready to capture
+    """
+    return robot_state.output_bit_register_64
```

Comment:
Could this be better named as `_capture_started`? The current name makes it sound like it will do something, i.e. start a capture.

Reply:
Same for `_finish_capture`

Reply:
ready_for_capture. finish_capture is not used

Reply:
Removed function that was not used (finish_capture and ready_to_capture)